### PR TITLE
Implement `U256InvModN` hint

### PR DESCRIPTION
**Pull request by Sh0g0-1758**
Fixes: #630

### CODE REVIEW

**Comment by Sh0g0-1758** [comment]
I took the tests from the examples given [here](https://github.com/sympy/sympy/blob/d91b8ad6d36a59a879cc70e5f4b379da5fdd46ce/sympy/core/intfunc.py#L362)

**Comment by Sh0g0-1758** [comment]
Following This implementation :
```
                    from starkware.python.math_utils import igcdex

                    b = {b0} + ({b1} << 128)
                    n = {n0} + ({n1} << 128)

                    (_, r, g) = igcdex(n, b)
                    if n == 1:
                        memory{g0_or_no_inv} = 1
                        memory{g1_option} = 0
                        memory{s_or_r0} = {b0}
                        memory{s_or_r1} = {b1}
                        memory{t_or_k0} = 1
                        memory{t_or_k1} = 0
                    elif g != 1:
                        if g % 2 == 0:
                            g = 2
                        s = b // g
                        t = n // g
                        memory{g0_or_no_inv} = g & 0xffffffffffffffffffffffffffffffff
                        memory{g1_option} = g >> 128
                        memory{s_or_r0} = s & 0xffffffffffffffffffffffffffffffff
                        memory{s_or_r1} = s >> 128
                        memory{t_or_k0} = t & 0xffffffffffffffffffffffffffffffff
                        memory{t_or_k1} = t >> 128
                    else:
                        r %= n
                        k = (r * b - 1) // n
                        memory{g0_or_no_inv} = 0
                        memory{s_or_r0} = r & 0xffffffffffffffffffffffffffffffff
                        memory{s_or_r1} = r >> 128
                        memory{t_or_k0} = k & 0xffffffffffffffffffffffffffffffff
                        memory{t_or_k1} = k >> 128
```

File: pkg/hintrunner/core/hint_test.go [hunk 1]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+		SOrR0Val.SetZero()
+
+		require.Equal(
+			t,
+			mem.MemoryValueFromFieldElement(SOrR0Val),
+			utils.ReadFrom(vm, VM.ExecutionSegment, 3),
+		)
+
+		SOrR1Val := &f.Element{}
+		_, err = SOrR1Val.SetString("1")
```
Comment (cicr99) [inline pkg/hintrunner/core/hint_test.go:522]: Same comment about the string

File: pkg/hintrunner/core/hint_test.go [hunk 2]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+		G1OptionVal.SetZero()
+
+		require.Equal(
+			t,
+			mem.MemoryValueFromFieldElement(G1OptionVal),
+			utils.ReadFrom(vm, VM.ExecutionSegment, 2),
+		)
+
+		SOrR0Val := &f.Element{}
+		_, err = SOrR0Val.SetString("1002")
```
Comment (cicr99) [inline pkg/hintrunner/core/hint_test.go:605]: Same comment about the string

File: pkg/hintrunner/core/hint_test.go [hunk 3]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+		SOrR1Val.SetZero()
+
+		require.Equal(
+			t,
+			mem.MemoryValueFromFieldElement(SOrR1Val),
+			utils.ReadFrom(vm, VM.ExecutionSegment, 4),
+		)
+
+		TOrK0Val := &f.Element{}
+		_, err = TOrK0Val.SetString("50")
```
Comment (cicr99) [inline pkg/hintrunner/core/hint_test.go:624]: Same comment about the string

File: pkg/hintrunner/core/hint_test.go [hunk 4]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+		G0OrNoInvVal.SetZero()
+
+		require.Equal(
+			t,
+			mem.MemoryValueFromFieldElement(G0OrNoInvVal),
+			utils.ReadFrom(vm, VM.ExecutionSegment, 1),
+		)
+
+		SOrR0Val := &f.Element{}
+		_, err = SOrR0Val.SetString("1")
```
Comment (cicr99) [inline pkg/hintrunner/core/hint_test.go:687]: Same comment about the string

File: pkg/hintrunner/core/hint_test.go [hunk 5]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+		SOrR1Val.SetZero()
+
+		require.Equal(
+			t,
+			mem.MemoryValueFromFieldElement(SOrR1Val),
+			utils.ReadFrom(vm, VM.ExecutionSegment, 4),
+		)
+
+		TOrK0Val := &f.Element{}
+		_, err = TOrK0Val.SetString("1")
```
Comment (cicr99) [inline pkg/hintrunner/core/hint_test.go:706]: Same comment about the string

File: pkg/hintrunner/core/hint_test.go [hunk 6]
```diff
@@ -455,6 +455,274 @@ func TestDivModDivisionByZeroError(t *testing.T) {
...
+			SOrR1:     SOrR1,
+			TOrK0:     TOrK0,
+			TOrK1:     TOrK1,
+		}
+
+		err := hint.Execute(vm, nil)
+		require.Nil(t, err)
+
+		G0OrNoInvVal := &f.Element{}
+		_, err = G0OrNoInvVal.SetString("1")
```
[the remark this answers is not in the document]
Comment (Sh0g0-1758) [reply pkg/hintrunner/core/hint_test.go]: done.

File: pkg/hintrunner/core/hint_test.go (hunk 1 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/core/hint_test.go]: done.

File: pkg/hintrunner/core/hint_test.go (hunk 2 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/core/hint_test.go]: done.

File: pkg/hintrunner/core/hint_test.go (hunk 3 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/core/hint_test.go]: done.

File: pkg/hintrunner/core/hint_test.go (hunk 4 above)
Comment (Sh0g0-1758) [reply pkg/hintrunner/core/hint_test.go]: done.

**Pull request merged**